src/core/TrainController.cpp

```diff
@@ -78,6 +114,22 @@ namespace godot {
         return mover;
     }
 
+    void TrainController::set_train_id(const String &p_train_id) {
+        train_id = p_train_id;
+    }
+
+    String TrainController::get_train_id() const {
+        if (train_id.is_empty()) {
+            String new_id = get_name().to_lower();
+            if (new_id.is_empty()) {
+                new_id = "train";
```

Comment:
Maybe change to something more like `train_default_id` to highlight the fact that this is the default id?

If someone would debug IDs, seeing this would be much convenient for future content makers to see that they got default train id